src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnectionBuilderExtensions.cs

```diff
@@ -28,28 +29,20 @@ public static IHubConnectionBuilder WithJsonProtocol(this IHubConnectionBuilder
 
         public static IHubConnectionBuilder WithLoggerFactory(this IHubConnectionBuilder hubConnectionBuilder, ILoggerFactory loggerFactory)
         {
-            hubConnectionBuilder.AddSetting(HubConnectionBuilderDefaults.LoggerFactoryKey,
-                loggerFactory ?? throw new ArgumentNullException(nameof(loggerFactory)));
+            if (loggerFactory == null)
+            {
+                throw new ArgumentNullException(nameof(loggerFactory));
+            }
+
+            hubConnectionBuilder.Services.AddSingleton(loggerFactory);
             return hubConnectionBuilder;
         }
 
         public static IHubConnectionBuilder WithLogger(this IHubConnectionBuilder hubConnectionBuilder, Action<ILoggerFactory> configureLogging)
```

Comment:
I don't want you to fix this now but we have a design issue by newing up the LoggerFactory where it never gets disposed.